### PR TITLE
feat(frontend): add related content to letters and add tools and letters to related contents

### LINKED ISSUE #1308: Améliorations page de modèle de courrier
Ces pages sont de loin les plus visitées du site, généralement par une entrée directe via les moteurs de recherche. Nous souhaitons les améliorer, notamment afin de permettre aux usagers de poursuivre leur navigation sur notre site pour consulter d'autres contenus.

To do : 
- Ajouter le module sur la droite avec les contenus associés
- Retirer la mention "Modèles de courrier fournis par vos services de renseignement des DIRECCTE en région" en bas de la page
- Retirer le wording "modèle de courrier" dans le titre du contenu

### LINKED ISSUE #1519: Mise en avant des outils et modèles de document dans la suggestion de contenus
Nous souhaitons mettre l'accent sur les contenus les plus actionnables de notre site, i.e. les modèles de document et les outils, dans les suggestions de contenus qui apparaissent sur la droite d'un contenu.

To do :
- on garde le principe de la requête sur le nom du document pour récupérer cette liste, mais on prend désormais 5 contenus
- les outils doivent être affichés en dessous du bloc "les articles pouvant vous intéresser" avec le design qu'ils ont sur la homepage (leur titre + un bouton d'action dans une card)
- idem pour les modèles, on les affiche de la même manière que les outils (en dessous des articles mais au-dessus des outils).
- pour ne pas surcharger la page, on met au max 1 modèle et 1 outil en plus des 3 articles, si la requête remonte plus de 1 modèle et 1 outil, alors on affiche uniquement l'outil le plus pertinent et le modèle le plus pertinent

### PR DESCRIPTION
**Pull request by UnbearableBear**
closes #1519 closes #1308 
Screenshot 2019-10-23 at 14 50 30

### CODE REVIEW

**Comment by UnbearableBear** [comment]
@nkrmr Désolé, pour adapter aux outils et lettres j'ai du revoir un peu le design !

**Comment by lionelB** [comment]
@UnbearableBear  on peut avoir la liste pour les documents et une carte pour le premier outil ? et que l'outils soit au debut ?

File: packages/code-du-travail-frontend/src/common/Answer.js [hunk 1]
```diff
@@ -84,29 +109,62 @@ function Answer({
...
+                .filter(link => link.title !== title)
+                .slice(0, 3)
+                .map(link => (
+                  <StyledListItem key={link.slug}>
+                    <Link
+                      href={`/${getRouteBySource(link.source)}/[slug]`}
+                      as={`/${getRouteBySource(link.source)}/${link.slug}`}
+                      passHref
+                    >
+                      <LargeLink variant="light">
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/common/Answer.js]: Can we keept the previous style for Article and add Tile block under ?

**Review by lionelB — CHANGES_REQUESTED** [review]
It would be nice a to add a snapshot on modele-courriers page taht check we have some related items

**Comment by UnbearableBear** [comment]
@lionelB modele-courriers page uses `Answer` but you're right, I'll add a test to ensure related items are displayed when `Answer` is given the correct data

File: packages/code-du-travail-frontend/src/common/Answer.js (hunk 1 above)
Comment (UnbearableBear) [reply packages/code-du-travail-frontend/src/common/Answer.js]: I followed the specs given by clement concerning the order but I'm going to make some proposals and see what we decide

**Comment by UnbearableBear** [comment]
@ClementChapalain Une préférence ? (Sur l'ordre aussi) 

Screenshot 2019-10-23 at 14 50 30
Screenshot 2019-10-23 at 20 12 05
Screenshot 2019-10-23 at 20 39 47

**Comment by ClementChapalain** [comment]
celui là ! si tu peux d'ailleurs en profiter pour virer la ligne à gauche des articles ce serait top je pense :)

image

**Comment by UnbearableBear** [comment]
Et voici le résultat final !
Screenshot 2019-10-24 at 14 39 06

File: packages/code-du-travail-frontend/src/common/__tests__/Answer.test.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import React from "react";
 import { render } from "@testing-library/react";
+import { SOURCES } from "@cdt/sources";
 
 import Answer from "../Answer";
 
 describe("<Answer />", () => {
-  it("should render", () => {
+  it("renders", () => {
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/common/__tests__/Answer.test.js:8]: all the tests are using it("should")  can we keep consistency here ?

File: packages/react-ui/src/Grid/index.js [hunk 3]
```diff
@@ -17,15 +18,15 @@ export const Grid = styled.ul`
 `;
 
 export const GridCell = styled.li`
-  flex-basis: calc(100% / 4 - 2 * ${spacing.small});
+  display: flex;
   flex-grow: 0;
   flex-shrink: 1;
+  width: calc(100% / 4 - 2 * ${spacing.small});
```
Comment (lionelB) [inline packages/react-ui/src/Grid/index.js:24]: it's not clear what was wrong with flex-basis that width solves  ?

**Review by lionelB — COMMENTED** [review]
It seems that you have added som ui component too but I don't see them, can you add some screenshot of what was added and what usage ?

File: packages/react-ui/src/Grid/index.js (hunk 3 above)
Comment (UnbearableBear) [reply packages/react-ui/src/Grid/index.js:24]: It solves nor changes absolutely nothing, I was tinkering with IE11 and forgot to put the old value back !

**Comment by UnbearableBear** [comment]
I did not add any component, however, I realized that the tile would always take 100% of the container width which is not always wanted so I added a `display: inline-flex` instead of a `display: flex` and made changes accordingly. I also mirrored the `Tile` props to the `LargeLink` prop concerning the `variant` prop. Finally I added a `size` prop to both the `Button` and the `Tile` component which can now have slightly smaller versions (used in the linked contents)

**Comment by UnbearableBear** [comment]
Screenshot 2019-10-24 at 19 07 47
Screenshot 2019-10-24 at 19 07 37
Screenshot 2019-10-24 at 19 07 29

**Review by lionelB — APPROVED** [review]

**Review by nkrmr — APPROVED** [review]

**Pull request merged**